File: BraintreeCore/src/main/java/com/braintreepayments/api/AnalyticsEvent.java
package com.braintreepayments.api;

import androidx.annotation.NonNull;
import androidx.room.ColumnInfo;
import androidx.room.Entity;
import androidx.room.PrimaryKey;

@Entity(tableName = "analytics_event")
public class AnalyticsEvent {

    @PrimaryKey(autoGenerate = true)
    @ColumnInfo(name = "_id")
    public int id;

    @NonNull
    @ColumnInfo(name = "name")
    private final String name;

    @ColumnInfo(name = "timestamp")
    private final long timestamp;

    AnalyticsEvent(@NonNull String name, long timestamp) {
        this.name = name;
        this.timestamp = timestamp;
    }

    @NonNull
    public String getName() {
        return name;
    }

    public long getTimestamp() {
        return timestamp;
    }
}

File: BraintreeCore/src/main/java/com/braintreepayments/api/AnalyticsEvent.kt
package com.braintreepayments.api

import androidx.annotation.RestrictTo

@RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
internal data class AnalyticsEvent constructor(
    val name: String,
    val payPalContextId: String? = null,
    val linkType: String? = null,
    val venmoInstalled: Boolean = false,
    val isVaultRequest: Boolean = false,
    val startTime: Long? = null,
    val endTime: Long? = null,
    val endpoint: String? = null,
    val timestamp: Long = System.currentTimeMillis(),
)
